firebase-firestore/src/main/java/com/google/firebase/firestore/local/QueryContext.java

```diff
@@ -0,0 +1,31 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.firebase.firestore.local;
+
+/** A tracker to keep a record of important details during database local query execution. */
+public class QueryContext {
```

Comment:
IMO the class name "QueryContext" is not great. To me, its name implies that it is used by the query execution as "input" or a way to acquire resources needed to perform a query. Since this is an internal class I'd name it exactly what it does, something like "DocumentReadCounter". The generality of naming it "QueryContext" is confusing to me.

Reply:
This class will be shared with other features, for example streaming counts. That's why it has a generic name.

Reply:
Discussed offline. I still feel that we should name this class according to its immediate purpose to improve readability (e.g. a name like "DocumentReadCounter" instead of "QueryContext"). If some future change wants to generalize its purpose then it can be easily refactored and renamed accordingly. But this won't block my approval, and would like to hear from @wu-hui .

Reply:
Fair enough. QueryContext it is.